File: spring-backend/src/main/java/br/dev/hygino/dto/ResponseBookDto.java
package br.dev.hygino.dto;

import java.util.UUID;

import br.dev.hygino.models.Book;

public record ResponseBookDto(
        UUID id,
        String title,
        String author,
        String personalCode,
        Integer edition,
        String publisher,
        Integer totalPages,
        String bookStatus) {

    public ResponseBookDto(Book obj) {
        this(
                obj.getId(),
                obj.getTitle(),
                obj.getAuthor(),
                obj.getPersonalCode(),
                obj.getEdition(),
                obj.getPublisher(),
                obj.getTotalPages(),
                obj.getBookStatus().name());
    }
}

File: backend/src/book/dto/response-book.dto.ts
import { Book } from "../entities/book.entity";

export class ResponseBookDto{
    id:number;
    title: string;
    publisher: string;
    edition: number;
    author: string;
    createdAt?: Date;
    updatedAt?: Date;

    constructor(book:Book){
        this.id = book.id;
        this.title = book.title;
        this.publisher = book.publisher;
        this.edition = book.edition;
        this.author = book.author?.name ;
        this.createdAt = book.createdAt;
        this.updatedAt = book.updatedAt;
    }
}